smart-contracts/test/Lock/erc721/safeTransferFrom.js

```diff
@@ -37,53 +34,24 @@ contract('Lock ERC721', accounts => {
       let ownerOf = await locks['FIRST'].ownerOf.call(ID)
       keyDataAfter = await locks['FIRST'].keyDataFor.call(to)
       assert.equal(ownerOf, to)
-      assert.equal(Web3Utils.toUtf8(keyDataAfter), '')
+      assert.equal(keyDataAfter, null)
     })
 
-    it('should work if some data is passed in ', async () => {
+    it('should work if some data is passed in', async () => {
       await locks['FIRST'].purchaseFor(accounts[7], Web3Utils.toHex('Julien'), {
         value: Units.convert('0.01', 'eth', 'wei'),
         from: accounts[7]
       })
       ID = await locks['FIRST'].getTokenIdFor.call(accounts[7])
-      let sender = Web3Utils.toChecksumAddress(accounts[7])
-      let receiver = Web3Utils.toChecksumAddress(accounts[6])
-      // Using encodeFunctionCall as a workaround until the upgrade to Truffle v5.x. Can't call overloaded functions from js currently...
-      let encodedCall = abi.encodeFunctionCall(
-        {
-          name: 'safeTransferFrom',
-          type: 'function',
-          inputs: [
-            {
-              type: 'address',
-              name: '_from'
-            },
-            {
-              type: 'address',
-              name: '_to'
-            },
-            {
-              type: 'uint256',
-              name: '_tokenId'
-            },
-            {
-              type: 'bytes',
-              name: 'data'
-            }
-          ]
-        },
-        [sender, receiver, Web3Utils.toHex(ID), Web3Utils.toHex('Julien')]
+      await locks['FIRST'].methods['safeTransferFrom(address,address,uint256,bytes)'](
+        accounts[7], accounts[6], Web3Utils.toHex(ID), Web3Utils.toHex('Julien'), { from: accounts[7] }
```

Comment:
Nice. I was hoping we could lose the `encodeFunctionCall` with this update.

Reply:
Yes, this is much nicer.  I would have saved it for a separate PR though... however the original method stopped working.  I'm unsure why